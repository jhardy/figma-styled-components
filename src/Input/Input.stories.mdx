import { Input } from "./Input";
import { Story, Meta } from "@storybook/addon-docs/blocks";

<Meta title="MDX|Input" component={Input} />

# Input

<Story name="Input">
  <div className="component-box">
    <Input placeholder="I am an input" />
  </div>
</Story>
