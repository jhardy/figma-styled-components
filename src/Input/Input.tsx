import styled from "styled-components";

export const Input = styled.input`
  display: flex;
  align-items: center;
  width: 100%;
  height: 30px;
  margin: 1px 0 1px 0;
  padding: 8px 4px 8px 7px;
  color: rgba(0, 0, 0, 0.8);
  border: 1px solid transparent;
  border-radius: 2px;
  outline: none;
  background-color: #ffffff;
  font-family: "Inter", sans-serif;
  font-weight: 400;
  font-size: 11px;
  line-height: 16px;
  letter-spacing: 0.005em;

  &:hover {
    color: rgba(0, 0, 0, 0.8);
    border: 1px solid rgba(0, 0, 0, 0.1);
  }

  &:active,
  &:focus {
    padding: 8px 4px 8px 6px;
    color: #000000;
    border: 2px solid #18a0fb;
    border-radius: 2px;
  }

  &::selection {
    color: #000000;
    background-color: rgba(24, 145, 251, 0.3);
  }

  &::placeholder {
    color: rgba(0, 0, 0, 0.3);
  }

  &:disabled {
    color: rgba(0, 0, 0, 0.3);
  }
`;
