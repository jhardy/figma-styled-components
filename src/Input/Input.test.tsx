import 'jest-styled-components'
import * as React from 'react'
import * as renderer from 'react-test-renderer'
import { Input } from './Input'



test('A default Input', () => {
  const tree = renderer.create(<Input />).toJSON()
  expect(tree).toMatchSnapshot()
})


test('A  Input with placeholder', () => {
  const tree = renderer.create(<Input placeholder='Hello' />).toJSON()
  expect(tree).toMatchSnapshot()
})


