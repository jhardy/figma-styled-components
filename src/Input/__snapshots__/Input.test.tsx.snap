// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`A  Input with Icon 1`] = `
.c2 {
  color: currentColor;
  line-height: 0;
}

.c1 {
  width: 32px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  color: rgba(0,0,0,0.5);
}

.c0 {
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: 100%;
  height: 32px;
  margin: 1px 0 1px 0;
  padding: 8px 4px 8px 0;
  color: rgba(0,0,0,0.8);
  border: 1px solid transparent;
  border-radius: 2px;
  outline: none;
  font-family: 'Inter',sans-serif;
  font-weight: 400;
  font-size: 11px;
  line-height: 16px;
  -webkit-letter-spacing: 0.005em;
  -moz-letter-spacing: 0.005em;
  -ms-letter-spacing: 0.005em;
  letter-spacing: 0.005em;
}

.c0 input {
  border: none;
  outline: none;
  width: 100%;
  height: 24px;
}

.c0:hover {
  color: rgba(0,0,0,0.8);
  box-shadow: inset 0 0 0 1px rgba(0,0,0,0.1);
}

.c0:active,
.c0:focus-within {
  color: #000000;
  box-shadow: inset 0 0 0 2px #18a0fb;
}

.c0::selection {
  color: #000000;
  background-color: rgba(24,145,251,0.3);
}

.c0::-webkit-input-placeholder {
  color: rgba(0,0,0,0.3);
}

.c0::-moz-placeholder {
  color: rgba(0,0,0,0.3);
}

.c0:-ms-input-placeholder {
  color: rgba(0,0,0,0.3);
}

.c0::placeholder {
  color: rgba(0,0,0,0.3);
}

.c0:disabled {
  color: rgba(0,0,0,0.3);
}

<div
  className="c0"
>
  <div
    className="icon-holder c1"
  >
    <div
      className="c2"
    >
      <svg
        fill="none"
        height={32}
        width={32}
      >
        <path
          clipRule="evenodd"
          d="M12 16.05V9h1v7.05a2.5 2.5 0 010 4.9V23h-1v-2.05a2.5 2.5 0 010-4.9zm2 2.45a1.5 1.5 0 11-3 0 1.5 1.5 0 013 0zm5 4.5h1v-7.05a2.5 2.5 0 000-4.9V9h-1v2.05a2.5 2.5 0 000 4.9V23zm2-9.5a1.5 1.5 0 10-3 0 1.5 1.5 0 003 0z"
          fill="currentColor"
          fillOpacity={0.8}
          fillRule="evenodd"
        />
      </svg>
    </div>
  </div>
  <input
    className=""
    placeholder="Hello"
    value="icon"
  />
</div>
`;

exports[`A  Input with placeholder 1`] = `
.c0 {
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: 100%;
  height: 32px;
  margin: 1px 0 1px 0;
  padding: 8px 4px 8px 8px;
  color: rgba(0,0,0,0.8);
  border: 1px solid transparent;
  border-radius: 2px;
  outline: none;
  font-family: 'Inter',sans-serif;
  font-weight: 400;
  font-size: 11px;
  line-height: 16px;
  -webkit-letter-spacing: 0.005em;
  -moz-letter-spacing: 0.005em;
  -ms-letter-spacing: 0.005em;
  letter-spacing: 0.005em;
}

.c0 input {
  border: none;
  outline: none;
  width: 100%;
  height: 24px;
}

.c0:hover {
  color: rgba(0,0,0,0.8);
  box-shadow: inset 0 0 0 1px rgba(0,0,0,0.1);
}

.c0:active,
.c0:focus-within {
  color: #000000;
  box-shadow: inset 0 0 0 2px #18a0fb;
}

.c0::selection {
  color: #000000;
  background-color: rgba(24,145,251,0.3);
}

.c0::-webkit-input-placeholder {
  color: rgba(0,0,0,0.3);
}

.c0::-moz-placeholder {
  color: rgba(0,0,0,0.3);
}

.c0:-ms-input-placeholder {
  color: rgba(0,0,0,0.3);
}

.c0::placeholder {
  color: rgba(0,0,0,0.3);
}

.c0:disabled {
  color: rgba(0,0,0,0.3);
}

<div
  className="c0"
>
  <input
    className=""
    placeholder="Hello"
    value="test"
  />
</div>
`;

exports[`A default Input 1`] = `
.c0 {
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: 100%;
  height: 32px;
  margin: 1px 0 1px 0;
  padding: 8px 4px 8px 8px;
  color: rgba(0,0,0,0.8);
  border: 1px solid transparent;
  border-radius: 2px;
  outline: none;
  font-family: 'Inter',sans-serif;
  font-weight: 400;
  font-size: 11px;
  line-height: 16px;
  -webkit-letter-spacing: 0.005em;
  -moz-letter-spacing: 0.005em;
  -ms-letter-spacing: 0.005em;
  letter-spacing: 0.005em;
}

.c0 input {
  border: none;
  outline: none;
  width: 100%;
  height: 24px;
}

.c0:hover {
  color: rgba(0,0,0,0.8);
  box-shadow: inset 0 0 0 1px rgba(0,0,0,0.1);
}

.c0:active,
.c0:focus-within {
  color: #000000;
  box-shadow: inset 0 0 0 2px #18a0fb;
}

.c0::selection {
  color: #000000;
  background-color: rgba(24,145,251,0.3);
}

.c0::-webkit-input-placeholder {
  color: rgba(0,0,0,0.3);
}

.c0::-moz-placeholder {
  color: rgba(0,0,0,0.3);
}

.c0:-ms-input-placeholder {
  color: rgba(0,0,0,0.3);
}

.c0::placeholder {
  color: rgba(0,0,0,0.3);
}

.c0:disabled {
  color: rgba(0,0,0,0.3);
}

<div
  className="c0"
>
  <input
    className=""
  />
</div>
`;
